gem/ci/build.ps1

```diff
@@ -118,39 +123,48 @@ function Clear-ContainerBuilds(
         Remove-Item $ENV:VOLUME_ROOT -Force -Recurse -ErrorAction Continue
     }
 
-    # this provides example data which ConvertFrom-String infers parsing structure with
-    $template = @'
+    Write-Host "`nPruning Volumes"
+    docker volume prune
+
+    Write-Host "`nPruning Networks"
+    docker network prune
+
+    if ($Name -ne $null)
+    {
+        # this provides example data which ConvertFrom-String infers parsing structure with
+        $template = @'
 {Version*:10.2.3*} {ID:5b84704c1d01} {[DateTime]Created:2019-02-07 18:24:51} +0000 GMT
 {Version*:latest} {ID:0123456789ab} {[DateTime]Created:2019-01-29 00:05:33} +0000 GMT
 '@
-    $output = docker images --filter=reference="$Namespace/${Name}" --format "{{.Tag}} {{.ID}} {{.CreatedAt}}"
-    Write-Output @"
+        $output = docker images --filter=reference="$Namespace/${Name}" --format "{{.Tag}} {{.ID}} {{.CreatedAt}}"
+        Write-Host @"
 
 Found $Namespace/${Name} images:
 $($output | Out-String)
 
 "@
 
-    if ($output -eq $null) { return }
-
-    Write-Output "Filtering removal candidates..."
-    # docker image prune supports filter until= but not repository like 'puppetlabs/foo'
-    # must use label= style filtering which is a bit more inconvenient
-    # that output is also not user-friendly!
-    # engine doesn't maintain "last used" or "last pulled" metadata, which would be more useful
-    # https://github.com/moby/moby/issues/4237
-    $output |
-      ConvertFrom-String -TemplateContent $template |
-      ? { $_.Created -lt $OlderThan } |
-      # ensure 'latest' are listed first
-      Sort-Object -Property Version -Descending |
-      % {
-        Write-Output "Removing Old $Namespace/${Name} Image $($_.Version) ($($_.ID)) Created On $($_.Created)"
-        $forcecli = if ($Force) { '-f' } else { '' }
-        docker image rm $_.ID $forcecli
-      }
-
-    Write-Output "`nPruning Dangling Images"
+        if ($output -eq $null) { return }
```

Comment:
If this is exiting the entire method, we should make sure we don't have cleanup calls below this that won't get executed.

Reply:
That was already there... just the indentation change makes the diff hard to read.

If no images were returned from our search then

* there won't be any to `docker image rm`
* there shouldn't be any dangling containers to prune


But I think I should refactor this out anyhow to a separate function so this logic is clearer - the branching obscures things.